### PR TITLE
feat: adding new badge variants and examples

### PR DESCRIPTION
**Pull request by TahimiLeonBravo**
fix: #1741 

## Changes proposed in this PR:
- add new variants to the Badge
- add custom badge examples

- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/Badge/readme.md [hunk 1]
```diff
@@ -43,6 +43,56 @@ import { Badge } from 'react-rainbow-components';
...
+```js
+import React from 'react';
+import { Badge } from 'react-rainbow-components';
+
+<div className="rainbow-p-vertical_large rainbow-align-content_center rainbow-flex_wrap">
+    <Badge
+        className="rainbow-m-around_medium"
+        label="Signal Badge"
+        title="the badge title"
+        style={{'color': 'white', 'backgroundColor': 'forestgreen', 'boxShadow': 'inset 0 0 0 1px'}}
```
Comment (TahimiLeonBravo) [inline src/components/Badge/readme.md:80]: we should not add the styles inline instead create a const

File: src/components/Badge/readme.md [hunk 2]
```diff
@@ -43,6 +43,56 @@ import { Badge } from 'react-rainbow-components';
...
+    />
+
+    <Badge
+        className="rainbow-m-around_medium"
+        label="Floating Badge"
+        title="the badge title"
+        style={{'color': 'white', 'backgroundColor': 'lightsalmon', 'borderColor': 'darksalmon', 'boxShadow': 'gray 0 6px 16px -6px'}}
```
Comment (TahimiLeonBravo) [inline src/components/Badge/readme.md:87]: ^^

File: src/components/Badge/readme.md [hunk 3]
```diff
@@ -43,6 +43,56 @@ import { Badge } from 'react-rainbow-components';
...
+    />
+
+    <Badge
+        className="rainbow-m-around_medium"
+        label="Desert Badge"
+        title="the badge title"
+        style={{'color': 'peru', 'backgroundColor': 'bisque',}}
```
Comment (TahimiLeonBravo) [inline src/components/Badge/readme.md:94]: ^^

File: src/components/Badge/index.js [hunk 4]
```diff
@@ -29,8 +29,17 @@ Badge.propTypes = {
...
+     * inverse, lightest, brand, outline-brand, warning, success and error. This value defaults to default. */
+    variant: PropTypes.oneOf([
+        'default',
+        'inverse',
+        'lightest',
+        'outline-brand',
+        'brand',
+        'warning',
+        'success',
+        'error',
```
Comment (yvmunayev) [inline src/components/Badge/index.js:41]: You need to add this props in the declarations file (index.d.ts)

**Pull request merged**